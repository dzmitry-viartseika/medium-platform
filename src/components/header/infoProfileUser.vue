<template>
  <div>
    <div class="app-info-profile">
      <template v-if="userPhoto">
        <img :src="userPhoto" alt="" class="app-info-profile__photo">
      </template>
      <template v-else>
        <div class="app-info-profile__circle"></div>
      </template>
      <div class="app-info-profile__firstname">
        {{ userFirstname }}
      </div>
    </div>
  </div>
</template>

<script>

import { get } from 'lodash';

export default {
  name: 'infoProfileUser',
  computed: {
    userInfo: {
      get() {
        return this.$store.getters.userInfo;
      },
      set(data) {
        this.$store.dispatch('setUserInfo', data);
      },
    },
    userPhoto() {
      return get(this.userInfo, 'image', '');
    },
    userFirstname() {
      return get(this.userInfo, 'username', '');
    },
  },
};
</script>

<style scoped lang="scss">
@import "../../assets/scss/variables";
  .app-info-profile {
    display: flex;
    align-items: center;
    cursor: default;

    img {
      max-width: 30px;
      height: auto;
      object-fit: cover;
    }

    &__circle {
      border-radius: 50%;
      width: 30px;
      height: 30px;
      background: $colorBtn;
    }
  }
</style>
