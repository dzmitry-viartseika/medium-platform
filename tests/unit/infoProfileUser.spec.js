import { createLocalVue, shallowMount } from '@vue/test-utils';
import infoProfileUser from '@/components/header/infoProfileUser.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);
const store = new Vuex.Store({});

describe('infoProfileUser.vue', () => {
  test('snapshot', () => {
    const $store = {
      state: {
        userInfo: 1,
      },
      getters: {
        userInfo: () => ({}),
      },
    };
    const wrapper = shallowMount(infoProfileUser, {
      localVue,
      store,
      mocks: {
        $store,
      },
    });
    const div = wrapper.find('.app-info-profile');
    expect(div.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
