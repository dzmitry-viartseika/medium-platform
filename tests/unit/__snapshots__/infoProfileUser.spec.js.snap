// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infoProfileUser.vue snapshot 1`] = `
<div>
  <div class="app-info-profile">
    <div class="app-info-profile__circle"></div>
    <div class="app-info-profile__firstname">

    </div>
  </div>
</div>
`;
